src/Mesh/Interpolation.jl

```diff
@@ -273,185 +553,686 @@ struct InterpolationCubedSphere{T <: Integer, FT <: AbstractFloat}
+        end
+
+        if f2==0 # applying phis
+            @inbounds vout_jk[tj,tk] *= (wb_sh[tj] / (ξ2l-m1_r_sh[tj]))
+        end
+        sync_threads() 
+            
+        if tj==1 # reduction
+            if f2==0
+                for ij in 2:qm1
+                    @inbounds vout_jk[1,tk]  += vout_jk[ij,tk]
+                end
+            else
+                if f2 ≠ 1
+                    @inbounds vout_jk[1,tk] = vout_jk[f2,tk]
+                end
+            end
+
+            if f3==0 # applying phit
+                @inbounds vout_jk[1,tk] *= (wb_sh[tk] / (ξ3l-m1_r_sh[tk]))
+            end
+        end
+        sync_threads()
+
+        if tj==1 && tk==1 # reduction
+            if f3==0
+                for ik in 2:qm1
+                    @inbounds vout_jk[1,1] += vout_jk[1,ik] 
+                end
+            else
+                if f3 ≠ 1
+                    @inbounds vout_jk[1,1] = vout_jk[1,f3]
+                end
+            end
+            @inbounds v[off + i, st_no] = vout_jk[1,1] * fc
+        end
+        #-----------------------------------
+    end
+    #-----------------------------------
+    return nothing
+end
+#--------------------------------------------------------
+"""
+    write_interpolated_data_cubed_sphere(intrp_cs::InterpolationCubedSphere{FT,T,FTV,FTVD,TVD,UI8AD,UI16V,UI16VD,I32V},
+                                              iv::AbstractArray{FT}, varnames, filename) where {FT <: AbstractFloat,
+                                                                                                 T <: Int,
+                                                                                               FTV <: AbstractVector{FT},
+                                                                                              FTVD <: AbstractVector{FT},
+                                                                                               TVD <: AbstractVector{T},
+                                                                                             UI8AD <: AbstractArray{UInt8,2},
+                                                                                             UI16V <: AbstractVector{UI16},
+                                                                                            UI16VD <: AbstractVector{UI16},
+                                                                                              I32V <: AbstractVector{Int32}}
+
+This interpolation function gathers interpolated data onto process # 0 and writed data to a NetCDF file.
+
+# input
+ - `intrp_cs` Initialized cubed sphere interpolation structure
+ - `iv` interpolated variables
+ - `varnames` Tuple of Interpolated variable name strings
+ - `filename` Filename of the NetCDF file to be written
+"""
+function write_interpolated_data_cubed_sphere(intrp_cs::InterpolationCubedSphere{FT,T,FTV,FTVD,TVD,UI8AD,UI16V,UI16VD,I32V},
```

Comment:
We can use the same function for both interpolations, and simply define two different methods. Also `FT` is the only type parameter used, so we only need that
```suggestion
function write_interpolated_data(intrp_cs::InterpolationCubedSphere{FT},
```